src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1574,45 +1574,33 @@ private String getTokenHash(String token) {
      * another call to acquireToken. It may try multi resource refresh token for
      * second attempt.
      */
-    private class RefreshItem {
-        String mRefreshToken;
-
-        String mKey;
-
-        boolean mMultiResource;
-
-        UserInfo mUserInfo;
-
-        String mRawIdToken;
-
+    private static class RefreshItem {
+        final String mRefreshToken;
+        final String mKey;
+        final boolean mMultiResource;
+        final UserInfo mUserInfo;
+        final String mRawIdToken;
+        final String mTenantId;
         String mKeyWithUserId;
-
         String mKeyWithDisplayableId;
-        
-        String mTenantId;
 
         public RefreshItem(String keyInCache, AuthenticationRequest request, TokenCacheItem item) {
+            if (item == null) {
+                throw new IllegalArgumentException("item");
```

Comment:
All what I'm saying is, I don't think we should throw unchecked exception in our code, instead, we can throw checked exception.